subsys/bluetooth/mesh/adv_legacy.c

```diff
@@ -234,7 +235,11 @@ void bt_mesh_adv_init(void)
 
 int bt_mesh_adv_enable(void)
 {
-	enabled = true;
+	/* The advertiser thread relies on BT_MESH_SUSPENDED flag. No point in starting the
+	 * advertiser thread if the flag is not set.
+	 */
+	__ASSERT_NO_MSG(ADV_ENABLED);
```

Comment:
Why assert here (considering asserts can be disabled), may be better to return -EADV?

Reply:
Why ever use asserts if they can be disabled?

Reply:
I guess the main purpose of assertion in assertion post action that is `weak` in Zephyr and customer is able to implement it on its own purpose. Generally, assertion is changed on hardware reset in final product as the only way for recovering device workability from unknown and unpredictable situation.
https://docs.zephyrproject.org/latest/kernel/services/other/fatal.html

Is this the real place where device requires hardware reset in "fields"? 

It seems @omkar3141 suggestion is more suitable. Since functional prototype returns operation status, it is better to return error code.